metrics/runtime_metrics_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package metrics
+
+import (
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/uber-go/tally"
+)
+
+func TestRuntimeReporter(t *testing.T) {
+	reporter := NewRuntimeReporter(tally.NoopScope, time.Millisecond)
+	assert.False(t, reporter.IsRunning())
+	reporter.Start()
+	assert.True(t, reporter.IsRunning())
+	runtime.GC()
+	time.Sleep(5 * time.Millisecond)
+	reporter.report()
+	reporter.Close()
+}
+
+func TestStartRuntimeReporter(t *testing.T) {
+	reporter := StartReportingRuntimeMetrics(tally.NoopScope, time.Millisecond)
+	assert.True(t, reporter.IsRunning())
+
+	runtime.GC()
+	time.Sleep(5 * time.Millisecond)
+	reporter.report()
+	time.Sleep(5 * time.Millisecond)
+	reporter.report()
+	reporter.Close()
```

Comment:
This was copied over but nothing magic about 5 afaik. Runtime metrics are reported every millsecond, even 1 millisecond should do it.